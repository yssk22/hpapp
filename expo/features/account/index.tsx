import ListItemClearCache from './internal/ListItemClearCache';

export { ListItemClearCache };
