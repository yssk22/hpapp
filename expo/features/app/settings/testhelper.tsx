import { act, render } from '@testing-library/react-native';

import { SettingsProvider, SettingsProviderProps } from './';
import SettingsList from './internals/SettingsList';

/**
 * Clear all settings
 */
export async function clearSettings() {
  await Promise.all(
    SettingsList.map((settings) => {
      settings.clear();
    })
  );
}

export async function renderSettingsProvider(children: React.ReactNode, props: SettingsProviderProps = {}) {
  await clearSettings();
  const content = render(<SettingsProvider {...props}>{children}</SettingsProvider>);
  await act(async () => {});
  return content;
}
