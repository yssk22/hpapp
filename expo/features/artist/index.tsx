import ArtistMemberFollowIcon from './internals/ArtistMemberFollowIcon';
import ArtistMemberIcon, { ArtistMemberIconSize } from './internals/ArtistMemberIcon';

export { ArtistMemberFollowIcon, ArtistMemberIcon, ArtistMemberIconSize };
