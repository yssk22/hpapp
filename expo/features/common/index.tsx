import AmebloIcon from './internals/AmebloIcon';
import AssetIcon from './internals/AssetIcon';
import CalendarDateIcon from './internals/CalendarDateIcon';
import ConsentGate from './internals/ConsentGate';
import ErrorBoundary, { FallbackComponent } from './internals/ErrorBoundary';
import ExternalImage from './internals/ExternalImage';
import Initializer from './internals/Initializer';
import Link from './internals/Link';
import Loading from './internals/Loading';
import Text from './internals/Text';
import useAssetContent from './internals/useAssetContent';
import useAsync from './internals/useAsync';
import useErrorMessage from './internals/useErrorMessage';
import useIsMounted from './internals/useIsMounted';
import useMutationPromise from './internals/useMutationPromise';
import useReloadableAsync, { ReloadableAysncResult } from './internals/useReloadableAsync';
import useURICache from './internals/useURICache';

export {
  AmebloIcon,
  AssetIcon,
  CalendarDateIcon,
  ConsentGate,
  ErrorBoundary,
  ExternalImage,
  FallbackComponent,
  Initializer,
  Link,
  Loading,
  Text,
  useErrorMessage,
  useMutationPromise,
  useReloadableAsync,
  ReloadableAysncResult,
  useAssetContent,
  useAsync,
  useIsMounted,
  useURICache
};
