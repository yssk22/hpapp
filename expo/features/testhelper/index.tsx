import {
  mockUseNavigation,
  renderUserScreen,
  renderUserComponent,
  renderGuestComponent,
  waitForUserRootRerendered
} from './internals/jest';
import { ScreenDecorator } from './internals/storybook';

export {
  mockUseNavigation,
  renderUserScreen,
  renderUserComponent,
  renderGuestComponent,
  waitForUserRootRerendered,
  ScreenDecorator
};
