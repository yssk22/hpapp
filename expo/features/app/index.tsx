import { maybeDev } from '@hpapp/foundation/environment';
import { LocalMediaManagerProvider } from '@hpapp/system/media';

import AppRelayProvider from './internals/AppRelayProvider';
import { default as Root, AppRootProps as RootProps } from './internals/AppRoot';
import AppUpdateBanner from './internals/AppUpdateBanner';
import { ThemeProvider, ThemeProviderProps } from './theme';

export type AppRootProps = RootProps & ThemeProviderProps;

export function AppRoot({ screens, ...props }: AppRootProps) {
  return (
    <ThemeProvider {...props}>
      <AppRelayProvider>
        <LocalMediaManagerProvider name={maybeDev() ? 'hellofanapp-dev' : 'hellofanapp'}>
          <AppUpdateBanner />
          <Root screens={screens} />
        </LocalMediaManagerProvider>
      </AppRelayProvider>
    </ThemeProvider>
  );
}
