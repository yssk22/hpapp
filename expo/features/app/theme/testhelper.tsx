import { SettingsProviderProps } from '@hpapp/features/app/settings';
import { renderSettingsProvider } from '@hpapp/features/app/settings/testhelper';

import { ThemeProvider } from './';

export async function renderThemeProvider(children: React.ReactNode, props: SettingsProviderProps = {}) {
  return renderSettingsProvider(<ThemeProvider>{children}</ThemeProvider>, props);
}
