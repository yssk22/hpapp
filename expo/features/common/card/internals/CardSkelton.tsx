import { useThemeSkeltonColor } from '@hpapp/features/app/theme';
import { Spacing } from '@hpapp/features/common/constants';
import { ThemeColorScheme } from '@hpapp/system/theme';
import ContentLoader, { Rect } from 'react-content-loader/native';

export type CardSkeltonProps = {
  colorScheme?: ThemeColorScheme;
};

export default function CardSkelton({ colorScheme = 'primary' }: CardSkeltonProps) {
  const primary = useThemeSkeltonColor(colorScheme);
  const innerBarX = Spacing.Small + 3 + Spacing.Medium + 50 + Spacing.Medium;
  const innerBarWidth = 400 - (Spacing.Small + 3 + Spacing.Medium + 50 + Spacing.Medium * 3);
  return (
    <ContentLoader speed={2} width="100%" height="225" viewBox="0 0 400 225" backgroundColor={primary}>
      <Rect x={Spacing.Small} y="0" rx="0" ry="0" width={400 - 2 * Spacing.Small} height="45" />
      <Rect x={Spacing.Small} y="225" rx="0" ry="0" width={400 - 2 * Spacing.Small} height="3" />

      <Rect x={Spacing.Small} y="45" rx="0" ry="0" width="3" height="180" />
      <Rect x={400 - Spacing.Small - 3} y="45" rx="0" ry="0" width="3" height="180" />

      <Rect x={Spacing.Small + 3 + Spacing.Medium} y="65" rx="0" ry="0" width="50" height="50" />
      <Rect x={Spacing.Small + 3 + Spacing.Medium} y="145" rx="0" ry="0" width="50" height="50" />

      <Rect x={innerBarX} y="75" rx="0" ry="0" width={innerBarWidth} height="5" />
      <Rect x={innerBarX} y="100" rx="0" ry="0" width={innerBarWidth} height="5" />

      <Rect x={innerBarX} y="155" rx="0" ry="0" width={innerBarWidth} height="5" />
      <Rect x={innerBarX} y="180" rx="0" ry="0" width={innerBarWidth} height="5" />
    </ContentLoader>
  );
}
