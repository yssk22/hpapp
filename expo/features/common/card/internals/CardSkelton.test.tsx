import { renderUserComponent } from '@hpapp/features/testhelper';
import { composeStories } from '@storybook/react';

import * as stories from './CardSkelton.stories';

describe('CardSkelton', () => {
  const composed = composeStories(stories);
  Object.keys(composed).forEach((key) => {
    const C = (composed as any)[key];
    test(key, async () => {
      const content = await renderUserComponent(<C uniqueKey="snapshots" animate={false} />);
      expect(content).toMatchSnapshot();
    });
  });
});
