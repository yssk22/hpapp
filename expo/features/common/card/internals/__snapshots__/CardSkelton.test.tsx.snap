// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSkelton Basic 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      fullScreenSwipeShadowEnabled={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="225"
                bbWidth="100%"
                focusable={false}
                height="225"
                interval={0.25}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {},
                    {
                      "flex": 0,
                      "height": 225,
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={225}
                vbWidth={400}
                width="100%"
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGRect
                    clipPath="snapshots-animated-diff"
                    fill={
                      {
                        "brushRef": "snapshots-diff",
                        "type": 1,
                      }
                    }
                    height="100%"
                    propList={
                      [
                        "fill",
                      ]
                    }
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <RNSVGDefs>
                    <RNSVGClipPath
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      name="snapshots-animated-diff"
                    >
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="45"
                        rx="0"
                        ry="0"
                        width={376}
                        x={12}
                        y="0"
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="3"
                        rx="0"
                        ry="0"
                        width={376}
                        x={12}
                        y="225"
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="180"
                        rx="0"
                        ry="0"
                        width="3"
                        x={12}
                        y="45"
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="180"
                        rx="0"
                        ry="0"
                        width="3"
                        x={385}
                        y="45"
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="50"
                        rx="0"
                        ry="0"
                        width="50"
                        x={31}
                        y="65"
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="50"
                        rx="0"
                        ry="0"
                        width="50"
                        x={31}
                        y="145"
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="5"
                        rx="0"
                        ry="0"
                        width={271}
                        x={97}
                        y="75"
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="5"
                        rx="0"
                        ry="0"
                        width={271}
                        x={97}
                        y="100"
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="5"
                        rx="0"
                        ry="0"
                        width={271}
                        x={97}
                        y="155"
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="5"
                        rx="0"
                        ry="0"
                        width={271}
                        x={97}
                        y="180"
                      />
                    </RNSVGClipPath>
                    <RNSVGLinearGradient
                      gradient={
                        [
                          0,
                          -3349518,
                          0.5,
                          -1118482,
                          1,
                          -3349518,
                        ]
                      }
                      gradientTransform={null}
                      gradientUnits={0}
                      name="snapshots-diff"
                      x1="-100%"
                      x2="0%"
                      y1={0}
                      y2={0}
                    />
                  </RNSVGDefs>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitle=""
        backTitleVisible={false}
        backgroundColor="#0075c2"
        color="white"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="/"
        titleColor="white"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</View>
`;
